main.js

```diff
@@ -129,13 +129,12 @@ function throttle(func, limit) {
 
 // Function to start MutationObserver
 const observe = () => {
+  const runDocumentMutations = throttle(async () => {
```

Comment:
while you're at it, this doesn't need to be `async`

```suggestion
  const runDocumentMutations = throttle(() => {
```